### PR TITLE
refactor(infra): migrate to new infra

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5562** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5562?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5595** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5595?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5497** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5497?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5563** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5563?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5564** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5564?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #5693** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5693?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #5617** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5617?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #5618** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5618?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #5565** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5565?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5565?utm_source=stack-comment).

File: packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx [hunk 1]
```diff
@@ -46,8 +46,7 @@ export const PageHeaderMenuButton = ({
 }: PageMenuProps) => {
   const t = useAFFiNEI18N();
 
-  // fixme(himself65): remove these hooks ASAP
```
Comment (pengx17) [inline packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx:49]: We did it!

File: packages/frontend/core/src/modules/infra-web/global-scope/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  const currentPageService = useServiceOptional(CurrentPageService, {
+    provider: workspaceProvider ?? ServiceCollection.EMPTY.provider(),
+  });
+
+  const pageProvider = useLiveData(
+    currentPageService?.currentPage ?? new LiveData<Page | null>(null)
+  )?.services;
+
+  console.log('global', pageProvider ?? workspaceProvider ?? rootProvider);
```
Comment (pengx17) [inline packages/frontend/core/src/modules/infra-web/global-scope/index.tsx]: Not sure if it is better to handle all of these in rxjs

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/modules/infra-web/global-scope/index.tsx (hunk 2 above)
Comment (EYHN) [reply packages/frontend/core/src/modules/infra-web/global-scope/index.tsx]: What do you mean, provide an example?

Comment (pengx17) [reply packages/frontend/core/src/modules/infra-web/global-scope/index.tsx]: Something like this, define a smart service provider:

```ts
class SmartServiceProvider {
  constructor(
    private readonly currentWorkspaceService: CurrentWorkspaceService,
    private readonly provider: ServiceProvider
  ) {}

  value = LiveData.from(
    this.currentWorkspaceService.currentWorkspace.pipe(
      switchMap(workspace => {
        const workspaceProvider = workspace?.services;
        const pageService = workspaceProvider?.get(CurrentPageService, {
          optional: true,
        });
        const currentPage = pageService?.currentPage;
        return (currentPage ?? new LiveData<Page | null>(null)).pipe(
          map(page => {
            return page?.services ?? workspaceProvider ?? this.provider;
          })
        );
      })
    ),
    this.provider
  );
}
```

and then `useService(SmartServiceProvider).value`

Comment (EYHN) [reply packages/frontend/core/src/modules/infra-web/global-scope/index.tsx]: Thank you for your suggestion, but after yesterday's discussion of the multi-tab mode.
I feel that the design of this global scope is not very practical. In the future, each part on the page will have its own scope. global scope may be removed in the future.

**Pull request merged**